service/src/main/java/bio/terra/cbas/dependencies/bard/BardServiceImpl.java

```diff
@@ -0,0 +1,91 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BardServiceImpl implements BardService, HealthCheck {
+
+  Logger log = LoggerFactory.getLogger(BardServiceImpl.class);
+  private static final String appId = "cbas";
+  private final BardClient bardClient;
+  private final BardServerConfiguration bardServerConfiguration;
+
+  public BardServiceImpl(BardClient bardClient, BardServerConfiguration bardServerConfiguration) {
+    this.bardClient = bardClient;
+    this.bardServerConfiguration = bardServerConfiguration;
+  }
+
+  private DefaultApi getDefaultAuthApi(BearerToken userToken) {
+    ApiClient client = bardClient.bardAuthClient(userToken);
+    return bardClient.defaultApi(client);
+  }
+
+  @Override
+  public void logRunSetEvent(
+      RunSetRequest request,
+      MethodVersion methodVersion,
+      List<String> workflowIds,
+      BearerToken userToken) {
+    String eventName = "workflow-submission";
+    HashMap<String, String> properties = new HashMap<>();
+    properties.put("runSetName", request.getRunSetName());
+    properties.put("methodName", methodVersion.method().name());
+    properties.put("methodSource", methodVersion.method().methodSource());
+    properties.put("methodVersionName", methodVersion.name());
+    properties.put("methodVersionUrl", methodVersion.url());
+    properties.put("recordCount", String.valueOf(request.getWdsRecords().getRecordIds().size()));
+    properties.put("workflowIds", workflowIds.toString());
+
+    Optional<GithubMethodDetails> maybeGitHubMethodDetails =
+        methodVersion.method().githubMethodDetails();
+    if (maybeGitHubMethodDetails.isPresent()) {
+      GithubMethodDetails githubMethodDetails = maybeGitHubMethodDetails.get();
+      properties.put("githubOrganization", githubMethodDetails.organization());
+      properties.put("githubRepository", githubMethodDetails.repository());
+      properties.put("githubIsPrivate", githubMethodDetails.isPrivate().toString());
+    }
+
+    logEvent(eventName, properties, userToken);
+  }
+
+  @Async("bardAsyncExecutor")
+  @Override
+  public void logEvent(String eventName, Map<String, String> properties, BearerToken userToken) {
+    if (bardServerConfiguration.enabled()) {
+      EventsEventLogRequest request = new EventsEventLogRequest().properties(properties);
+      try {
+        getDefaultAuthApi(userToken).eventsEventLog(eventName, appId, request);
+      } catch (Exception e) {
+        log.warn("Error logging event {} ", eventName, e);
+      }
+    }
+  }
+
+  @Override
+  public Result checkHealth() {
+    try {
+      ApiClient client = bardClient.apiClient();
+      DefaultApi defaultApi = bardClient.defaultApi(client);
+      defaultApi.systemStatus();
+      return new Result(true, "Ok");
```

Comment:
Is there anything in the `systemStatus()` response that could indicate a bad state? Or a message we could forward along instead of `OK`?

Reply:
Right now it looks like Bard will only return an error code if it's status is unhealthy, so there should not be a case where it returns a 200 from the status endpoint with a message that it's in a bad state (but I can confirm with the team)!